src/components/mc/cuda/kernel/mc_cuda_reduce.cu

```diff
@@ -157,29 +187,50 @@ ucc_status_t ucc_mc_cuda_reduce(const void *src1, const void *src2, void *dst,
     if (MC_CUDA_CONFIG->reduce_num_blocks != UCC_ULUNITS_AUTO) {
         bk = ucc_min(bk, MC_CUDA_CONFIG->reduce_num_blocks);
     }
-    switch (dt)
-    {
-        case UCC_DT_INT16:
-            DT_REDUCE_INT(int16_t, op, src1, src2, dst, count, stream, bk, th);
-            break;
-        case UCC_DT_INT32:
-            DT_REDUCE_INT(int32_t, op, src1, src2, dst, count, stream, bk, th);
-            break;
-        case UCC_DT_INT64:
-            DT_REDUCE_INT(int64_t, op, src1, src2, dst, count, stream, bk, th);
-            break;
-        case UCC_DT_FLOAT16:
-            ucc_assert(2 == sizeof(__half));
-            DT_REDUCE_FLOAT(__half, op, src1, src2, dst, count, stream, bk, th);
-            break;
-        case UCC_DT_FLOAT32:
-            ucc_assert(4 == sizeof(float));
-            DT_REDUCE_FLOAT(float, op, src1, src2, dst, count, stream, bk, th);
-            break;
-        case UCC_DT_FLOAT64:
-            ucc_assert(8 == sizeof(double));
-            DT_REDUCE_FLOAT(double, op, src1, src2, dst, count, stream, bk, th);
-            break;
+    switch (dt) {
+    case UCC_DT_INT16:
+        DT_REDUCE_INT(int16_t, op, src1, src2, dst, count, stream, bk, th);
+        break;
+    case UCC_DT_INT32:
+        DT_REDUCE_INT(int32_t, op, src1, src2, dst, count, stream, bk, th);
+        break;
+    case UCC_DT_INT64:
+        DT_REDUCE_INT(int64_t, op, src1, src2, dst, count, stream, bk, th);
+        break;
+    case UCC_DT_FLOAT16:
+        ucc_assert(2 == sizeof(__half));
```

Comment:
lets remove this assert, cuda half which comes from cuda_fp16.h is always 2 bytes as far as i can tell.

Reply:
I removed it, and also removed this assert in mc_reduce_multi.cu and mc_reduce_multi_alpha.cu